zbuild.h

```diff
@@ -95,6 +99,13 @@
 #  define z_size_t size_t
 #endif
 
+/* In zlib-compat some functions and types use unsigned long, but zlib-ng use size_t */
+#if defined(ZLIB_COMPAT)
+#  define z_ulong_t unsigned long
```

Comment:
I thought `zng_size_t` would be misleading as it's not supposed to have anything to do with `size_t` when using `ZLIB_COMPAT` and it could be confusing to contributors which one to use if the names are too similar.

Reply:
`zng_size_t` name would be explained by it being `size_t` when building zng, but something else whenever when building zlib-compatible.
`z_ulong_t` has the same kind of confusion but in the opposite direction.

Alternatively we could do `zng_huge_t` or `zng_biggest_t`, that way the name does not suggest any specific type except that it is something big.

Reply:
The type in legacy zlib was `uLong`, so I chose the closest that still conforms to our own naming style... I'm still waiting for better suggestions... Changing the name to final choice is pretty easy task using `git diff` as a way to ensure all instances are changed correctly.

One possible choice is also `z_max_t`,  which would follow the style used in `stdint.h`.

Reply:
I like `zng_max_`, that would probably be the least confusing naming, especially since it also follows stdint naming.

Reply:
Or perhaps `zng_uintmax_t` to include the type as well, also closely matching stdint type `uintmax_t`

Reply:
> Or perhaps `zng_uintmax_t` to include the type as well, also closely matching stdint type `uintmax_t`

If we can find a compromise that you, @nmoinvaz and I can agree is the best, I will rename it and we can merge this and move on.